federation-st/src/test/java/no/vegvesen/ixn/federation/FederationSystemST.java

```diff
@@ -12,40 +12,134 @@
+		try {
+			consumer.close();
+		} catch (JMSException ignore) {
+		}
+		try {
+			sinkSpTwo.close();
+		} catch (Exception ignore) {
+		}
+		sourceSpOne.close();
 
-		System.out.println("max " + stats.getMax());
-		System.out.println("min " + stats.getMin());
-		System.out.println("avg " + stats.getAverage());
 
+		LongStream latencies = latencyStatisticsBuilder.build();
+		LongSummaryStatistics stats = latencies.summaryStatistics();
 		System.out.println("latencies " + stats);
+		System.out.println("ttlOffsets " + ttlOffsetBuilder.build().summaryStatistics());
+		System.out.printf("Number of reconnects producer %d, consumer %d %n", this.reconnectsProducer, this.reconnectsConsumer);
+		long numberOfMessagesReceived = stats.getCount();
+		System.out.printf("Number of messages received %d, lost %d %n", numberOfMessagesReceived, numberOfMessagesToSend - numberOfMessagesReceived);
+	}
+
+	private MessageConsumer reconnectConsumer(Sink sinkSpTwo, MessageConsumer consumer) {
+		System.out.println("Reconnecting to sp-two queue in 30 seconds");
+		try {
+			consumer.close();
+			consumer = null;
+		} catch (JMSException ignore) {
+		}
+		for (int y = 0; y < 10 && consumer == null; y++) {
+			try {
+				Thread.sleep(30000);
+				System.out.printf("Reconnect attempt %d %n", y);
+				consumer = sinkSpTwo.createConsumer();
+			} catch (InterruptedException | NamingException | JMSException ex) {
+				System.out.println("Reconnect consumer " + ex.getMessage());
+			}
+		}
+		if (consumer == null) {
+			System.out.println("Could not reconnect to sp-two");
+			System.exit(-1);
+		}
+		reconnectsConsumer++;
+		System.out.println("Reconnected to sp-two queue");
+		return consumer;
+	}
+
+	private void sendOneMessageReconnect(Source sourceSpOne, int i, long timeToLive) throws InterruptedException, NamingException, JMSException {
+		String messageText = "message " + i + " " + randomString(1024 * 200);
+		try {
+			sourceSpOne.send(messageText, "NO", timeToLive);
+		} catch (JMSException e) {
+			reconnectProducer(sourceSpOne);
+			sourceSpOne.send(messageText, "NO", timeToLive);
+		}
+	}
+
+	private void reconnectProducer(Source sourceSpOne) throws InterruptedException, NamingException, JMSException {
+		System.out.println("Reconnecting to onramp in 30 seconds");
+		try {
+			sourceSpOne.close();
+		} catch (Exception ignore) {
+		}
+		Thread.sleep(30000);
+		for (int y = 0; y < 10 && !sourceSpOne.isConnected(); y++) {
+			System.out.printf("Reconnect source attempt %d %n", y);
+			sourceSpOne.start();
+		}
+		if (!sourceSpOne.isConnected()) {
+			System.out.println("Could not reconnect to sp-one");
+			System.exit(-2);
+		}
+		reconnectsProducer++;
+	}
+
+	public String randomString(int length) {
+		int leftLimit = 48; // numeral '0'
+		int rightLimit = 122; // letter 'z'
+		Random random = new Random();
+		return random.ints(leftLimit, rightLimit + 1)
+				.filter(i -> (i <= 57 || i >= 65) && (i <= 90 || i >= 97))
```

Comment:
Why are these values as they are? For inctance, why is the lower bound for i 57, when the leftLimit is 48? The same goes for the other side. Also, has to run several random.next()'s which will be filtered out both at both ends and the middle.

Reply:
I took inspiration from code that generated readable characters. I don'n need this limitation on printed characters.